### PR TITLE
Add moco batch shuffle

### PR DESCRIPTION
**Pull request by philippmwirth**
# Add moco batch shuffle

- Added a universal interface to load models from checkpoints in `lightly.models._loader.py`. Can load:
  - SimCLR checkpoint to SimCLR
  - SimCLR checkpoint to MoCo
  - MoCo checkpoint to MoCo
  
  Examples:
  ```python
   # load simclr from simclr
   model = ResNetSimCLR()
   model.load_from_state_dict(simclr_state_dict)

   # load moco from moco
   model = ResNetMoCo()
   model.load_from_state_dict(moco_state_dict)


   # load moco from simclr
   model = ResNetMoCo()
   model.load_from_state_dict(simclr_state_dict, strict=False)

   ```
- Added SplitBatchNorm to simulate multi-gpu behaviour.
- Added batch shuffling to MoCo for improved performance.

### CODE REVIEW

File: lightly/models/batchnorm.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                self.bias,
+                False,
+                self.momentum, 
+                self.eps
+            )
+        
+        return result
+
+
+def BatchNorm2d(num_features: int, num_splits: int, **kw):
```
Comment (IgorSusmelj) [inline lightly/models/batchnorm.py:83]: Maybe rename this? I'm afraid this could lead to misinterpretation and people mistake `nn.BatchNorm2d` from PyTorch with this replacement block. What do you think?

**Review by IgorSusmelj — CHANGES_REQUESTED** [review]

**Review by IgorSusmelj — APPROVED** [review]
